### PR TITLE
Bugfixes and improvements.

### PR DESCRIPTION
**Pull request by deathbaba**

### CODE REVIEW

**Comment by deathbaba** [comment]
@dkorolev you can take a look if you have some time. I agree that use gtest is a good idea.

File: src/posix/http_client_curl.cc [hunk 1]
```diff
@@ -112,7 +112,7 @@ bool HTTPClientPlatformWrapper::RunHTTPRequest() {
     cmd += "--data-binary @" + body_file_ + " ";
   }
 
-  cmd += "'" + url_requested_ + "'";
+  cmd += url_requested_;
```
Comment (dkorolev) [inline src/posix/http_client_curl.cc]: Looks wrong to me.

File: tests/Makefile [hunk 2]
```diff
@@ -1,25 +1,23 @@
 .PHONY: run all indent clean
 
 CPP=g++
-CPPFLAGS=-std=c++11 -Wall -W
+CPPFLAGS=-std=c++11 -Wall -W -Igoogletest/include -Igoogletest -Wno-missing-field-initializers
```
Comment (dkorolev) [inline tests/Makefile]: In December we ported GoogleTest to not require any extra `-I`-s.

Comment (deathbaba) [inline tests/Makefile]: I remember it, but this version does not require any patches.

**Pull request merged**